core/node/dns.go

```diff
@@ -0,0 +1,27 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	config "github.com/ipfs/go-ipfs-config"
+	doh "github.com/libp2p/go-doh-resolver"
+	madns "github.com/multiformats/go-multiaddr-dns"
+)
+
+func DNSResolver(cfg *config.Config) (*madns.Resolver, error) {
+	var opts []madns.Option
+	if url := cfg.DNS.DefaultResolver; url != "" {
+		if !strings.HasPrefix(url, "https://") {
```

Comment:
@lidel do we currently need to add support for non-DoH resolvers aside from using the built-in Go (i.e. OS-based) resolver as the default resolver?

Reply:
No, we only need DoH. Other things like [DoT](https://en.wikipedia.org/wiki/DNS_over_TLS) can be added if needed, but not needed atm.